src/lib/pure_pursuit/PurePursuit.cpp

```diff
@@ -34,21 +34,43 @@
 #include "PurePursuit.hpp"
 #include <mathlib/mathlib.h>
 
+
+PurePursuit::PurePursuit(ModuleParams *parent) : ModuleParams(parent)
+{
+	_param_handles.lookahead_gain = param_find("PP_LOOKAHD_GAIN");
+	_param_handles.lookahead_max = param_find("PP_LOOKAHD_MAX");
+	_param_handles.lookahead_min = param_find("PP_LOOKAHD_MIN");
+	updateParams();
+}
+
+void PurePursuit::updateParams()
+{
+	param_get(_param_handles.lookahead_gain, &_params.lookahead_gain);
+	param_get(_param_handles.lookahead_max, &_params.lookahead_max);
+	param_get(_param_handles.lookahead_min, &_params.lookahead_min);
+
+	ModuleParams::updateParams();
+
+}
+
 float PurePursuit::calcDesiredHeading(const Vector2f &curr_wp_ned, const Vector2f &prev_wp_ned,
 				      const Vector2f &curr_pos_ned,
-				      const float lookahead_distance)
+				      const float vehicle_speed)
 {
 	// Check input validity
-	if (!curr_wp_ned.isAllFinite() || !curr_pos_ned.isAllFinite() || lookahead_distance < FLT_EPSILON
-	    || !PX4_ISFINITE(lookahead_distance) || !prev_wp_ned.isAllFinite()) {
+	if (!curr_wp_ned.isAllFinite() || !curr_pos_ned.isAllFinite() || vehicle_speed < 0.f
```

Comment:
```suggestion
	if (!curr_wp_ned.isAllFinite() || !curr_pos_ned.isAllFinite() || vehicle_speed < FLT_EPSILON
```

Reply:
I want to allow passing a zero velocity to this function without potentially breaking things due to returning a NaN value. 
A zero speed value might be sent if the rover is coming to a stop, of crouse the pure pursuit algorithm doesn't do anything at that point but i'd still rather not return a Nan value.

Reply:
Okay, but then I would check `vehicle_speed < -FLT_EPSILON` to make sure to not catch 0.

Reply:
implemented in [a633f7f](https://github.com/PX4/PX4-Autopilot/commit/a633f7f7ca92cca5611d9a4ffbfab08f43751c12)